src/Infrastructure.Styles.Analyzer/RMSYN1102SeparateExitConditionsAnalyzer.cs

```diff
@@ -0,0 +1,142 @@
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Infrastructure.Styles.Analyzer
+{
+  [DiagnosticAnalyzer(LanguageNames.CSharp)]
+  public class RMSYN1102SeparateExitConditionsAnalyzer : DiagnosticAnalyzer
+  {
+    public const string DiagnosticId = "RMSYN1102";
+
+    internal const string Title = "Exit conditions must use separate IF statements instead of OR'ed statements";
+    internal const string Message = "There must not be OR'ed exit conditions";
+
+    internal const string Description =
+        "Exit conditions must use separate IF statements instead of OR'ed statements";
+
+    internal const string Category = "Readability";
+
+    internal static readonly IReadOnlyCollection<SyntaxKind> InvalidStatements = new Collection<SyntaxKind>()
+    {
+        SyntaxKind.ReturnStatement
+    };
+
+    private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+        id: DiagnosticId,
+        title: Title,
+        messageFormat: Message,
+        category: Category,
+        defaultSeverity: DiagnosticSeverity.Warning,
+        isEnabledByDefault: true,
+        description: Description);
+
+    public override void Initialize (AnalysisContext context)
+    {
+      context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+      context.EnableConcurrentExecution();
+      context.RegisterSyntaxNodeAction(AnalyzeLogicalExpression, SyntaxKind.LogicalOrExpression);
+      context.RegisterSyntaxNodeAction(AnalyzeLogicalExpression, SyntaxKind.OrPattern);
+    }
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
+      ImmutableArray.Create(Rule);
+
+    private static void AnalyzeLogicalExpression (SyntaxNodeAnalysisContext context)
+    {
+      var baseNode = context.Node;
+      SyntaxNode orNode;
+      if (baseNode.IsKind(SyntaxKind.LogicalOrExpression))
+        orNode = baseNode;
+      else if (baseNode.IsKind(SyntaxKind.OrPattern))
+        orNode = baseNode.Parent!;
+      else
+        return;
+
+      if (orNode.Parent == null)
+        return;
+
+      var parentIf = GetParentIfStatementOrNull(orNode);
+      if (parentIf is null)
+        return;
+
+      if (!IsLegalIfNode(parentIf))
+        CreateDiagnostic(context, parentIf.GetLocation());
+    }
+
+    private static bool IsLegalIfNode (IfStatementSyntax? ifNode)
+    {
+      return IsLegalIfStatement(ifNode!);
+    }
+
+    private static void CreateDiagnostic (SyntaxNodeAnalysisContext context, Location location)
+    {
+      var diagnostic = Diagnostic.Create(Rule, location);
+      context.ReportDiagnostic(diagnostic);
+    }
+
+
+    private static IfStatementSyntax? GetParentIfStatementOrNull (SyntaxNode node)
+    {
+      var ancestors = node.Ancestors();
+      ancestors = ancestors.SkipWhile(
+          n => n.IsKind(SyntaxKind.OrPattern) || n.IsKind(SyntaxKind.IsPatternExpression) ||
```

Comment:
Style: the operators are at the start of the line.

Reply:
This could be an easy-to-do syntax-checker (no quick-fix needed) and applies to all "chaining" such as booleans, math and concat (string+) operators. -> please ticket